### PR TITLE
proposal: sidecarset imagepullsecrets

### PR DESCRIPTION
**Pull request by veophi**
Signed-off-by: veophi <vec.g.sun@gmail.com>

### Ⅰ. Describe what this PR does
proposal: SidecarSet imagePullSecrets

### CODE REVIEW

**Review by furykerry — COMMENTED** [review]
/lgtm

File: docs/proposals/20210713-sidecarsetimagepullsecrects.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+In `SidecarSet` part, we only consider injecting their `imagePullSecrets` feilds into Pod.
+Users should manually  distribute the required `Secrets` to all the namespaces that the `SidecarSet` may be instantiated.
+
+### API Definition
+We add `imagePullSecrets` field in `apis/apps/v1alpha1/sidecarset_types.go`:
+```go
+type SidecarSetSpec struct {
+	...
+	// List of the names of secrets required by pulling sidecar container images
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
```
Comment (FillZpp) [inline docs/proposals/20210713-sidecarsetimagepullsecrects.md:48]: `corev1.LocalObjectReference` just has a field `name`, where is the namespace of the secret?

File: docs/proposals/20210713-sidecarsetimagepullsecrects.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  selector:
+    ...
+  updateStrategy:
+    ...
+  containers:
+    - name: my-sidecar
+      image: my-repository/centos:6.7
+     ...
+  imagePullSecrets:
+    - name: my-secret
```
Comment (FillZpp) [inline docs/proposals/20210713-sidecarsetimagepullsecrects.md:70]: where is the namespace of the secret?

**Pull request merged**